core/src/main/java/org/infinispan/commands/FlagAffectedCommand.java

```diff
@@ -1,14 +1,78 @@
 package org.infinispan.commands;
 
+import java.util.Arrays;
+import java.util.Set;
+
+import org.infinispan.commons.util.EnumUtil;
+import org.infinispan.context.Flag;
+
 /**
- * Commands affected by Flags should carry them over to the remote nodes.
- *
- * By implementing this interface the remote handler will read them out and restore in context;
- * flags should still be evaluated in the InvocationContext.
+ * Commands affected by Flags will be checked locally to control certain behaviors such whether or not to invoke
+ * certain commands remotely, check cache store etc.
  *
- * @author Sanne Grinovero <sanne@hibernate.org> (C) 2011 Red Hat Inc.
- * @since 5.0
+ * @author William Burns
+ * @since 6.0
```

Comment:
Here Will's comments seem more accurate, but I'd still keep the `@since 5.0` tag.